sheet.go

```diff
@@ -1167,8 +1176,31 @@ func (f *File) ProtectSheet(sheet string, settings *FormatSheetProtection) error
 		Sheet:               true,
 		Sort:                settings.Sort,
 	}
+
 	if settings.Password != "" {
-		ws.SheetProtection.Password = genSheetPasswd(settings.Password)
+		// settings.AlgorithmName == "SHA-512"
+		if settings.AlgorithmName == hashAlgorithmSHA512 {
+			err, result := genISOPasswordHash(saltHashData{
```

Comment:
I suggest that make `genSheetPasswd` support password generation with the different crypto algorithm, and it can be called in `ProtectSheet` and `UnprotectSheet`, since that password generation should be implemented in the same scope, the caller just as a wrapper.

Reply:
`genSheetPasswd` only requires one password param generate one XOR result password, but `genISOPasswordHash` requires 3(AlgorithmName, password, SpinCount) and return 4 (algorithmName, saltValue, hashValue and spinCount). I think it will be more complicated when these two method become one. Let's keep it separated entil more new algorithm need to be implemented.